src/reporters/humanReporter.ts

```diff
@@ -115,94 +126,114 @@ export class HumanReporter {
         runTime: number;
         stackTrace: string | null;
       }) => {
-        const msg = elem.stackTrace
-          ? `${elem.message}\n${elem.stackTrace}`
-          : elem.message;
+        if (
+          !concise ||
+          elem.outcome === ApexTestResultOutcome.Fail ||
+          elem.outcome === ApexTestResultOutcome.CompileFail
+        ) {
+          const msg = elem.stackTrace
+            ? `${elem.message}\n${elem.stackTrace}`
+            : elem.message;
 
-        testRowArray.push({
-          name: elem.fullName,
-          outcome: elem.outcome,
-          msg: elem.message ? msg : '',
-          runtime:
-            elem.outcome !== ApexTestResultOutcome.Fail ? `${elem.runTime}` : ''
-        });
+          testRowArray.push({
+            name: elem.fullName,
+            outcome: elem.outcome,
+            msg: elem.message ? msg : '',
+            runtime:
+              elem.outcome !== ApexTestResultOutcome.Fail
+                ? `${elem.runTime}`
+                : ''
+          });
+        }
       }
     );
 
-    let testResultTable = os.EOL.repeat(2);
-    testResultTable += tb.createTable(
-      testRowArray,
-      [
-        {
-          key: 'name',
-          label: nls.localize('testNameColHeader')
-        },
-        { key: 'outcome', label: nls.localize('outcomeColHeader') },
-        { key: 'msg', label: nls.localize('msgColHeader') },
-        { key: 'runtime', label: nls.localize('runtimeColHeader') }
-      ],
-      nls.localize('testResultsHeader')
-    );
+    let testResultTable: string;
+    if (testRowArray.length > 0) {
```

Comment:
With this change, we don't want to print this table at all if all tests passed, so added a conditional check.